Gems/Atom/Component/DebugCamera/Code/Source/NoClipControllerComponent.cpp

```diff
@@ -404,6 +402,36 @@ namespace AZ
             m_currentHeading = NormalizeAngle(m_currentHeading);
         }
 
+        void NoClipControllerComponent::SetCameraStateForward(float value)
+        {
+            m_inputStates[CameraKeys::Forward] = value < -deadZone ? false : true; 
```

Comment:
The condition can be removed by just flipping the comparison:

````
m_inputStates[CameraKeys::Forward] = (value >= -deadZone);
````

Similarly all the conditions of the other functions can be removed by using the right comparison directly.